src/main/java/com/chat/yourway/repository/ContactRepository.java

```diff
@@ -5,13 +5,12 @@
 
 import java.util.Optional;
 
-/**
- * {@link ContactRepository}
- *
- * @author Dmytro Trotsenko on 7/26/23
- */
 public interface ContactRepository extends JpaRepository<Contact, Integer> {
 
-    Optional<Contact> findByEmail(String email);
+  Optional<Contact> findByEmail(String email);
+
+  boolean existsByEmail(String email);
+
+  boolean existsByUsername(String username);
```

Comment:
Since you use search by specific fields here, could you also add indexes for this fields at the database side? (If it already exists - skip this comment)